eras/shelley/impl/src/Cardano/Ledger/Shelley/Rewards.hs

```diff
@@ -451,117 +456,159 @@ aggregateRewards pp rewards =
+  Natural ->
   Coin ->
+  Map (KeyHash 'StakePool crypto) (PoolRewardInfo crypto) ->
+  Map (Credential 'Staking crypto) (KeyHash 'StakePool crypto) ->
+  (Credential 'Staking crypto) ->
+  Coin ->
+  Coin
+computeMemberRewards pp_mv totalStake poolRIs ds cred (Coin c) = fromMaybe (Coin 0) $ do
+  pid <- Map.lookup cred ds
+  rewardInfo <- Map.lookup pid poolRIs
+  if notPoolOwner cred rewardInfo
+    then Nothing
+    else
+      Just $
+        memberRew
+          (poolPot rewardInfo)
+          (poolPs rewardInfo)
+          (StakeShare (c % (unCoin totalStake)))
+          (poolRelativeStake rewardInfo)
+  where
+    notPoolOwner (KeyHashObj hk) pri = hk `Set.notMember` (_poolOwners . poolPs $ pri)
+    notPoolOwner (ScriptHashObj _) _ = HardForks.allowScriptStakeCredsToEarnRewards pp_mv
+
+-- | The stake pool member reward calculation
+rewardOnePoolMember ::
   Natural ->
+  Coin ->
+  Set (Credential 'Staking c) ->
+  PoolRewardInfo c ->
+  Credential 'Staking c ->
+  Coin ->
+  Maybe (Set (Reward c))
+rewardOnePoolMember
+  pp_mv
+  (Coin totalStake)
+  addrsRew
+  rewardInfo
+  hk
+  (Coin c) =
+    if hk `Set.member` addrsRew && notPoolOwner hk && r /= Coin 0
+      then Just . Set.singleton $ Reward MemberReward (_poolId pool) r
+      else Nothing
+    where
+      pool = poolPs rewardInfo
+      sigma = poolRelativeStake rewardInfo
+      poolR = poolPot rewardInfo
+      tot = fromIntegral totalStake
+      r = memberRew poolR pool (StakeShare (fromIntegral c % tot)) sigma
+      notPoolOwner (KeyHashObj cred) = cred `Set.notMember` (_poolOwners . poolPs $ rewardInfo)
+      notPoolOwner (ScriptHashObj _) = HardForks.allowScriptStakeCredsToEarnRewards pp_mv
+
+-- | Calculate single stake pool specific values for the reward computation.
+mkPoolRewardInfo ::
+  (UnitInterval, NonNegativeInterval, Natural) ->
+  Coin ->
+  BlocksMade c ->
   Natural ->
-  PoolParams c ->
   Stake c ->
-  Rational ->
-  Rational ->
+  Map (Credential 'Staking c) (KeyHash 'StakePool c) ->
   Coin ->
-  Set (Credential 'Staking c) ->
-  ProvM
-    (Map (KeyHash 'StakePool c) (RewardProvenancePool c))
-    m
-    (Map (Credential 'Staking c) (Set (Reward c)))
-rewardOnePool
-  (pp_d, pp_a0, pp_nOpt, pp_mv)
-  -- (Decentralization parameter, Pool influence, Desired number of pools, major version)
+  Coin ->
+  PoolParams c ->
+  Either StakeShare (PoolRewardInfo c)
+mkPoolRewardInfo
+  (pp_d, pp_a0, pp_nOpt)
   r
-  blocksN
+  blocks
   blocksTotal
-  pool
-  (Stake stake)
-  sigma
-  sigmaA
+  stake
+  delegs
   (Coin totalStake)
-  addrsRew = modifyM (Map.insert key provPool) >> pure rewards'
+  (Coin activeStake)
+  pool = case Map.lookup (_poolId pool) (unBlocksMade blocks) of
+    Nothing -> Left (StakeShare sigma)
+    Just blocksN ->
+      let Coin pledge = _poolPledge pool
+          pledgeRelative = fromIntegral pledge % fromIntegral totalStake
+          sigmaA = if activeStake == 0 then 0 else fromIntegral pstakeTot % fromIntegral activeStake
```

Comment:
These are already `Integer`s, so `fromIntegral` is not needed
```suggestion
          pledgeRelative = pledge % totalStake
          sigmaA = if activeStake == 0 then 0 else pstakeTot % activeStake
```

Reply:
doh! good catch, thank you!